### PR TITLE
Ports Bay's wall-mounted air pumps

### PR DESCRIPTION
**Pull request by Arokha**
These are wall-mounted tank-holders that have a mask in them. Drag onto yourself to wear the mask and get air in emergencies.

That's cool and all, but the real gem in my opinion is the N2O version for surgery, so you don't need to literally attach a tank to your patient to operate on them.

Changed OpTable's throwpass to 1. It's a table. Like other tables. You can throw things over it (also required to be able to reach over it to apply a mask).

NOTE: You'd have to map these in. They just 'exist' in this PR.

NOTE NOTE: Forgot to mention, these aren't free air, they have a normal air tank inside them. You can use a screwdriver to open them and swap out the tank (swap out the O2 tanks with anesthetics for shenanigans ;D)

image

(Surgery kit sprite in another PR)

### CODE REVIEW

**Comment by Arokha** [comment]
Yes, one person at a time. It's got a mask in it, basically it's a mask on a little hose, for emergency use. Like I said, I really think the N2O ones are the star of this more than anything. Having a patient lay down for surgery and just dragging the mask onto them is way easier than "here let me equip this tank into your back slot" (what??)

If you end up not using them for anything, not really a big deal, you'd have to map them in for them to show up anywhere. But maybe consider the N2O ones.

File: code/game/machinery/oxygen_pump.dm [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	contained = new mask_type (src)
+
+/obj/machinery/oxygen_pump/Destroy()
+	if(breather)
+		breather.internal = null
+		if(breather.internals)
+			breather.internals.icon_state = "internal0"
+	if(tank)
+		qdel(tank)
+	if(breather)
```
[the remark this answers is not in the document]
Comment (Arokha) [reply code/game/machinery/oxygen_pump.dm]: That's interesting. This is just a flat copy-paste from Bay, where it's like that. I'm not sure why. Let me see if there's a real reason.

**Comment by Arokha** [comment]
(I should note that Cyantime has expressed interest in making a mobile version of the N2O one on wheels for legit anesthetic equipment OR use)

**Comment by Anewbe** [comment]
Roger roger.

File: code/game/machinery/oxygen_pump.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Arokha) [reply code/game/machinery/oxygen_pump.dm]: I didn't see a reason, so committed an improvement to the Destroy()

**Pull request merged**